### PR TITLE
set time-to-live for webpush messages

### PR DESCRIPTION
**Pull request by bwindels**
Configurable with a default of 15 minutes.

### CODE REVIEW

File: sygnal/webpushpushkin.py [hunk 1]
```diff
@@ -119,6 +120,12 @@ def __init__(self, name, sygnal, config):
         self.vapid_contact_email = self.get_config("vapid_contact_email")
         if not self.vapid_contact_email:
             raise PushkinSetupException("'vapid_contact_email' not set in config")
+        self.ttl = self.get_config("ttl")
+        if self.ttl:
+            if not isinstance(self.ttl, int):
+                raise PushkinSetupException("'ttl' must be an int if set")
+        else:
+            self.ttl = DEFAULT_TTL
```
Comment (callahad) [inline sygnal/webpushpushkin.py:128]: Would it be cleaner to do this?

```python
self.ttl = self.get_config("ttl", DEFAULT_TTL)
if not isinstance(self.ttl, int):
    raise PushkinSetupException("'ttl' must be an int if set")
```

**Comment by callahad** [comment]
Do we need to document the `ttl` configuration anywhere? (The VAPID config isn't documented either, so I guess we can let this slip, too, while WebPush is experimental)

Does `ttl` need to be added to the `UNDERSTOOD_CONFIG_FIELDS` list?

**Comment by bwindels** [comment]
Thanks for the review @callahad, ready for another look.

**Comment by callahad** [comment]
r+ once CI passes and if you promise that you've tested this locally and it at least appears to work.

**Review by callahad — APPROVED** [review]

**Comment by bwindels** [comment]
@callahad Have tested again, all seems to work :+1:

**Pull request merged**